servers/shared-snowplow-consumer/src/snowplow/shareableListItem/shareableListItemEventHandler.ts

```diff
@@ -24,79 +23,59 @@ export class ShareableListItemEventHandler extends EventHandler {
    */
   private static generateShareableListItemEvent(
-    data: ShareableListItemEventPayloadSnowplow,
+    data: ShareableListItemEventBridgePayload,
   ): ObjectUpdate {
     return {
-      schema: shareableListItemEventSchema.objectUpdate,
-      data: {
-        trigger: data.eventType,
-        object: 'shareable_list_item',
-      },
+      trigger: data['detail-type'],
+      object: 'shareable_list_item',
     };
   }
 
   private static generateEventContext(
-    data: ShareableListItemEventPayloadSnowplow,
+    data: ShareableListItemEventBridgePayload,
   ): SelfDescribingJson[] {
     return [
-      ShareableListItemEventHandler.generateSnowplowShareableListItemEvent(
-        data,
-      ),
+      createShareableListItem(
+        ShareableListItemEventHandler.generateSnowplowShareableListItemEvent(
+          data.detail.shareableListItem,
+        ),
+      ) as unknown as SelfDescribingJson,
     ];
   }
 
   /**
    * Static method to generate an object that maps properties received in the event payload object to the snowplow shareable_list_item object schema.
    */
   private static generateSnowplowShareableListItemEvent(
-    data: ShareableListItemEventPayloadSnowplow,
+    data: ShareableListItemEventBridgePayload['detail']['shareableListItem'],
   ): ShareableListItem {
-    const snowplowEvent = {
-      schema: shareableListItemEventSchema.shareable_list_item,
-      data: {
-        shareable_list_item_external_id:
-          data.shareable_list_item.shareable_list_item_external_id,
-        shareable_list_external_id:
-          data.shareable_list_item.shareable_list_external_id,
-        given_url: data.shareable_list_item.given_url,
-        title: data.shareable_list_item.title
-          ? data.shareable_list_item.title
-          : undefined,
-        excerpt: data.shareable_list_item.excerpt
-          ? data.shareable_list_item.excerpt
-          : undefined,
-        image_url: data.shareable_list_item.image_url
-          ? data.shareable_list_item.image_url
-          : undefined,
-        authors: data.shareable_list_item.authors
-          ? data.shareable_list_item.authors
-          : undefined,
-        publisher: data.shareable_list_item.publisher
-          ? data.shareable_list_item.publisher
-          : undefined,
-        note: data.shareable_list_item.note
-          ? data.shareable_list_item.note
-          : undefined,
-        sort_order: data.shareable_list_item.sort_order,
-        created_at: data.shareable_list_item.created_at,
-        updated_at: data.shareable_list_item.updated_at
-          ? data.shareable_list_item.updated_at
-          : undefined,
-      },
+    return {
+      shareable_list_item_external_id: data.shareable_list_item_external_id,
+      shareable_list_external_id: data.shareable_list_external_id,
+      given_url: data.given_url,
+      title: data.title ? data.title : undefined,
```

Comment:
same as my other comment regarding conditional undefined setters -- while we're here let's see if it's necessary

Reply:
Fixed!